### DISCUSSION TITLE
Поле algo

### DISCUSSION THREAD

**Discussion by idvlab** [RoboTradeCode/configurator#26] [General]
Как настроит поле algo в общем шаблоне?

**Comment by 81xx** [comment]
> Как настроит поле algo в общем шаблоне?

Уточни пожалуйста вопрос.

**Comment by 81xx** [comment]
> Как настроит поле algo в общем шаблоне?

А лучше сделать пример что есть сейчас и что хочешь сделать.

**Comment by idvlab** [comment]
https://configurator.robotrade.io/ftx/1?only_new=false вот здесь в поле algo указано spread_bot_cpp, я хотел бы в зависимости от алгоритма, который используется в этом сочетании exchange/instance указывать свое название алгоритма, например, test_core_php

**Comment by 81xx** [comment]
Как я понимаю сейчас нельзя, добавим тогда.

**Comment by khanbekov** [comment]
может быть добавить файл конфигурации этих полей? 

Как я это вижу: в папке binance/1/, рядом с assets.txt будет лежать файл config.json, в котором можно указать поля.

Сейчас их эти поля не изменить, да.

**Comment by 81xx** [comment]
Создал issue по этому вопросу - https://github.com/RoboTradeCode/configurator/issues/27

**Comment by khanbekov** [comment]
Начал работать над этой задачей.

 Я думаю, в файле настроек должны лежать только следующие поля (закомментированные **не** нужно выносить в _header.json_):

```js
{
  // "event": "config",
  "exchange": "ftx",
  "node": "configurator",
  "instance": "1",
  // "action": "send_full_config",
  // "message": null,
  "algo": "spread_bot_cpp"
}
```
Я предлагаю не выносить в настройки поля `action` и `message`, т.к. их динамически устанавливает Configurator. Аналогично, я полагаю, поле `event` тоже устанавливается самой программой.

Согласно логике работы Configurator (и его роли в торговой системе)  внутри _configs/_ лежат статичные настройки, которые меняются редко. Т.е. конфигурации Gate и Core, их ассеты, настройки алгоритма и т.п. 

Распространяя эту логику на файл _header.json_, в нём должны быть поля, которые относятся к настройкам торгового сервера (Core + Gate + Agent), соответственно это `exchange`, `instance`, `algo`.

 Поле `node` также лучше вынести в глобальные настройки Configurator, но не обязательно, можно оставить его в  _header.json_.

**Reply by 81xx** [reply]
Смотри, моя логика была в том, что бы сделать конфигуратор максимально гибким, т.е. что бы все можно было настроить вне кода. Давай  сделаем как ты предложил, но тогда надо это обязательно отметить в документации с теп же примером. Что бы те кто будут использовать понимали что есть фиксированные и не изменяемые поля.

**Reply by khanbekov** [reply]
Тогда я вынесу эти поля в _config.toml_, его можно будет редактировать. При каждом обращении к API Configurator будет считывать данные из конфига. Так мы добъемся и гибкости, и разделения глобальных и локальных настроек.

_config.toml_ я также опишу в вики

**Reply by 81xx** [reply]
Если ты про закоммментированные поля, то я за! Хорошая идея.
